quinn/src/recv_stream.rs

```diff
@@ -112,15 +115,17 @@ impl RecvStream {
         Poll::Ready(Ok(buf.filled().len()))
     }
 
-    /// Attempts to read from the stream into buf.
+    /// Attempt to read from the stream into the provided buffer, which may be uninitialized
     ///
-    /// On success, returns Poll::Ready(Ok(())) and places data in
-    /// the buf. If no data was read and the buffer had a non null capacity,
-    /// it implies that EOF has been reached.
+    /// On success, returns `Poll::Ready(Ok(num_bytes_read))` and places data into `buf`. If this
+    /// returns zero bytes read (and `buf` has a non-zero length), that indicates that the remote
+    /// side has [`finish`]ed the stream and the local side has already read all bytes.
```

Comment:
This needs to be reworded to account for returning a `()` rather than a `usize` on success.